### PR TITLE
BMA

### PR DESCRIPTION
**Pull request by amallia**

### CODE REVIEW

File: include/pisa/query/algorithm/block_max_ranked_and_query.hpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <vector>
+#include "query/queries.hpp"
+
+namespace pisa {
+
+struct block_max_ranked_and_query {
+
+    typedef bm25 scorer_type;
```
Comment (elshize) [inline include/pisa/query/algorithm/block_max_ranked_and_query.hpp:10]: `using scorer_type = bm25`

File: include/pisa/query/algorithm/block_max_ranked_and_query.hpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    block_max_ranked_and_query(uint64_t k) : m_topk(k) {}
+
+    template <typename CursorRange>
+    uint64_t operator()(CursorRange &&cursors, uint64_t max_docid) {
+        using Cursor = typename std::decay_t<CursorRange>::value_type;
+
+        m_topk.clear();
+        if (cursors.empty())
```
Comment (elshize) [inline include/pisa/query/algorithm/block_max_ranked_and_query.hpp:19]: Parens around `break`

File: include/pisa/query/algorithm/block_max_ranked_and_query.hpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                    i         = 1;
+                }
+            }
+            else {
+
+                ordered_cursors[0]->docs_enum.next();
+                uint64_t next_jump = ordered_cursors[0]->docs_enum.docid();
+
+                // Next jump is the last identifier in the current block arrangement
+                for (size_t j = 1; j < ordered_cursors.size(); ++j) {
```
Comment (elshize) [inline include/pisa/query/algorithm/block_max_ranked_and_query.hpp]: What's a `j`?

**Review by elshize — CHANGES_REQUESTED** [review]

**Comment by JMMackenzie** [comment]
Just committed some new attempts at this. It still doesn't quite work properly as far as I can tell (it can occasionally miss a document that should be scored). However, it generally seems closer to working than before. Will revisit soon.

**Pull request closed without merging**